spanner/request_id_header.go

```diff
@@ -137,7 +138,12 @@ func gRPCCallOptionsToRequestID(opts []grpc.CallOption) (reqID requestID, found
 func (wr *requestIDHeaderInjector) interceptUnary(ctx context.Context, method string, req, reply any, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
 	// It is imperative to search for the requestID before the call
 	// because gRPC's internals will consume the headers.
-	reqID, foundRequestID := gRPCCallOptionsToRequestID(opts)
+	mdWithRequestID, reqID, foundRequestID := gRPCCallOptionsToRequestID(opts)
+	if !foundRequestID {
+		panic(fmt.Sprintf("missing %q request header for %q", xSpannerRequestIDHeader, method))
```

Comment:
Do we want to panic here? or do a noop?

Reply:
You know I wondered about that too earlier, but a panic allows us to proactively find any bugs.

Reply:
I think logging error/warning should be good here, since we don't to stop the world for our customers because of tracing?

Reply:
I'll just then insert a TODO that can then be picked up for as we please later on.

Reply:
Done, thanks!